services/api/service.go

```diff
@@ -386,6 +405,74 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(opts blockSimOptions) error {
+	t := time.Now()
+	simErr := api.blockSimRateLimiter.send(opts.ctx, opts.req, opts.highPrio)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.currentCounter(),
+	})
+	if simErr != nil {
+		log.WithError(simErr).Error("block validation failed")
+		return simErr
+	}
+	log.Info("block validation successful")
+	return nil
+}
+
+// Demotes a set of builders who have matching collateral IDs. This function
+// continues to try demoting even if an error is encountered for any builder.
+func (api *RelayAPI) demoteBuildersByCollateralID(builderPubkey string) {
+	// Fetch builder collateral_id.
+	builder, err := api.db.GetBlockBuilderByPubkey(builderPubkey)
+	if err != nil {
+		api.log.WithError(err).Error("unable to get builder from database")
+	}
+	collateralID := builder.CollateralID
+
+	// Fetch additional builder pubkeys using the collateral_id.
+	allBuilders, err := api.db.GetBlockBuildersFromCollateralID(collateralID)
+	if err != nil {
+		api.log.WithError(err).Error("unable to get builders from collateral id")
+	}
+
+	// Demote all the pubkeys in both redis and the db.
+	for _, b := range allBuilders {
+		pk := b.BuilderPubkey
```

Comment:
consider labelling the variable as `pubkey` (instead of `pk`) for consistency

Reply:
done